src/extras/BitmapText.js

```diff
@@ -543,58 +543,61 @@ export default class BitmapText extends core.Container
      *
      * @static
      * @param {XMLDocument} xml - The XML document data.
-     * @param {PIXI.Texture|PIXI.Texture[]} textures - List of textures for each page.
+     * @param {Object.<string, PIXI.Texture>|PIXI.Texture|PIXI.Texture[]} textures - List of textures for each page.
+     *  If providing an object, the key is the `<page>` element's `file` attribute in the FNT file.
      * @return {Object} Result font object with font, size, lineHeight and char fields.
      */
     static registerFont(xml, textures)
     {
         const data = {};
         const info = xml.getElementsByTagName('info')[0];
         const common = xml.getElementsByTagName('common')[0];
-        const fileName = xml.getElementsByTagName('page')[0].getAttribute('file');
-        const res = getResolutionOfUrl(fileName, settings.RESOLUTION);
+        const pages = xml.getElementsByTagName('page');
+        const res = getResolutionOfUrl(pages[0].getAttribute('file'), settings.RESOLUTION);
+        const pagesTextures = {};
 
         data.font = info.getAttribute('face');
         data.size = parseInt(info.getAttribute('size'), 10);
         data.lineHeight = parseInt(common.getAttribute('lineHeight'), 10) / res;
         data.chars = {};
-        if (!(textures instanceof Array))
+
+        // Single texture, convert to list
+        if (textures instanceof core.Texture)
         {
             textures = [textures];
         }
 
+        // Convert the input Texture, Textures or object
+        // into a page Texture lookup by "id"
+        for (let i = 0; i < pages.length; i++)
+        {
+            const id = pages[i].getAttribute('id');
+            const file = pages[i].getAttribute('file');
+
+            pagesTextures[id] = textures[file] || textures[i];
```

Comment:
Should this be `file ? textures[file] : textures[i]` or something similar in the off chance that someone has a file with the exact same name as a property or method of arrays?